### ISSUE TITLE
Cannot use operators defined in internal types.

### ISSUE THREAD

**Issue by teo-tsirpanis** [dotnet/fsharp#16762]
I discovered that the compiler will not let me use an operator in an internal type, even from an internal function. I'm not sure if it's intentional, so opening it here as a bug.

**Repro steps**

```fsharp
open System

module internal M =
    type C(value: int) =
        static member inline (||+) (x: C, y: C) = 0
    
let inline internal f (x1: M.C) (x2: M.C) = x1 ||+ x2
```

**Expected behavior**

The snippet compiles successfully.

**Actual behavior**

```
error FS0043: The member or object constructor 'op_BarBarPlus' is not public. Private members may only be accessed from within the declaring type. Protected members may only be accessed from an extending type and cannot be accessed from inner lambda expressions.
```

**Known workarounds**

If I make the module public the snippet will compile successfully. I can do it in my use case (an F# library distributed in source form) and explicitly document that the types may become internal at any time.

**Related information**

Provide any related information (optional):

* .NET SDK 8.0.201

**Comment by KevinRansom** [comment]
This seems like the wrong behaviour.

The only way to make this compile is to decorate the module as public.  The error message identifies the operator as private, and yet the only active scope limiter is the module which is decorated as internal.

![image](https://github.com/dotnet/fsharp/assets/5175830/f897dd8c-690e-4a3c-8803-4e1ec89591f7)

**Comment by KevinRansom** [comment]
@dsyme clearly F# has had this behaviour for ever, do we really want to restrict operator type member visibility to public members.

![image](https://github.com/dotnet/fsharp/assets/5175830/02904dc1-2136-40c2-9be4-2f8e4d366b36)

**Comment by dsyme** [comment]
There's an approved language design suggestion for this, and https://github.com/dotnet/fsharp/pull/6805 addressed the issue, but that PR is quite old now.
